### PR TITLE
V1.3: Implement new `sub` command

### PR DESCRIPTION
**Pull request by ZHTang29**
Substitution command will search for and display a list of ingredient substitutions for the ingredient that was searched for.

Functionality
- [X] Implement Case-Insensitive Substitution
- [x] Implement Case-Sensitive Substitution
- [x] Implement Substitution that does not display repeated results
- [x] Add in pre-loaded list of substitutions for condiments 
- [x] Sub command should search through pre-loaded list then the recipe book before displaying results

Testing
- [x] Ensure existing tests are working
- [x] Write new tests for Substitution (wip for v1.4)

Documentation
- [x] Write Javadoc for new classes/methods

### CODE REVIEW

File: src/main/java/seedu/recipe/logic/parser/RecipeBookParser.java [hunk 1]
```diff
@@ -6,15 +6,7 @@
...
-import seedu.recipe.logic.commands.AddCommand;
-import seedu.recipe.logic.commands.ClearCommand;
-import seedu.recipe.logic.commands.Command;
-import seedu.recipe.logic.commands.DeleteCommand;
-import seedu.recipe.logic.commands.EditCommand;
-import seedu.recipe.logic.commands.ExitCommand;
-import seedu.recipe.logic.commands.FindCommand;
-import seedu.recipe.logic.commands.HelpCommand;
-import seedu.recipe.logic.commands.ListCommand;
+import seedu.recipe.logic.commands.*;
```
Comment (SeeuSim) [inline src/main/java/seedu/recipe/logic/parser/RecipeBookParser.java:9]: wildcard import

File: src/main/java/seedu/recipe/logic/commands/SubCommand.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        HashSet<Ingredient> subList = new HashSet<>();
+
+        // query all recipes in the current recipe book
+        for (Recipe r : recipeObservableList) {
+            HashMap<Ingredient, IngredientInformation> tokens = r.getIngredients();
+            // if recipe contains queried ingredient
+            if (tokens.containsKey(queryIngredient)) {
+                //inquire for subs
+                IngredientInformation rInfo = tokens.get(queryIngredient);
+                if (!rInfo.getSubstitutions().isEmpty()) {
```
Comment (SeeuSim) [inline src/main/java/seedu/recipe/logic/commands/SubCommand.java:57]: can be simplified to one statement, addAll from an empty list will not do anything

```java
subList.addAll(recipeInfo.getSubstitutions());
```

Comment (ZHTang29) [reply src/main/java/seedu/recipe/logic/commands/SubCommand.java:57]: Amended

File: src/main/java/seedu/recipe/logic/parser/RecipeBookParser.java (hunk 1 above)
Comment (ZHTang29) [reply src/main/java/seedu/recipe/logic/parser/RecipeBookParser.java:9]: done, thanks for checking

File: src/main/java/seedu/recipe/logic/commands/SubCommand.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        // query all recipes in the current recipe book
+        for (Recipe r : recipeObservableList) {
+            HashMap<Ingredient, IngredientInformation> tokens = r.getIngredients();
+            // if recipe contains queried ingredient
+            if (tokens.containsKey(queryIngredient)) {
+                //inquire for subs
+                IngredientInformation rInfo = tokens.get(queryIngredient);
+                subList.addAll(rInfo.getSubstitutions());
+            }
+        }
```
Comment (filbertphang) [inline src/main/java/seedu/recipe/logic/commands/SubCommand.java:69]: SLAP: the logic of `execute` can be abstracted out into another method

**Review by alson001 — APPROVED** [review]
LGTM. Just need to fix some checkstyles.

**Pull request merged**